src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs

```diff
@@ -0,0 +1,182 @@
+        /// if the requested text is at multilines, multiple sizes will be returned for each text located in a separate line. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
+        /// </summary>
+        /// <param name="textEditor">The TextEditor control containing the text.</param>
+        /// <param name="start">The start index of the text to get the size for</param>
+        /// <param name="end">The end index of the text to get the size for.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static List<Size2D> GetTextSize(TextEditor textEditor, int start, int end)
+        {
+            ValidateRange(start, end);
+
+            List<Size2D> list = GetSizeListFromNativeVector(Interop.TextEditor.GetTextSize(textEditor.SwigCPtr, (uint)start, (uint)end));
+            CheckSWIGPendingException();
+            return list;
+        }
+
+        /// <summary>
+        /// Get the rendered size of the text between start and end (included). <br />
+        /// if the requested text is at multilines, multiple sizes will be returned for each text located in a separate line. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
+        /// </summary>
+        /// <param name="textField">The TextField control containing the text.</param>
+        /// <param name="start">The start index of the text to get the size for</param>
+        /// <param name="end">The end index of the text to get the size for.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static List<Size2D> GetTextSize(TextField textField, int start, int end)
+        {
+            ValidateRange(start, end);
+
+            List<Size2D> list = GetSizeListFromNativeVector(Interop.TextField.GetTextSize(textField.SwigCPtr, (uint)start, (uint)end));
+            CheckSWIGPendingException();
+            return list;
+        }
+
+        /// <summary>
+        /// Get the rendered size of the text between start and end (included). <br />
+        /// if the requested text is at multilines, multiple sizes will be returned for each text located in a separate line. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
+        /// </summary>
+        /// <param name="textLabel">The TextLabel control containing the text.</param>
+        /// <param name="start">The start index of the text to get the size for</param>
```

Comment:
TextLabel could be include markup tag such as `<font>`, 
if Text was including markup, the index is including markup text ? or not?

for example, 
`"<font family='SamsungSans' weight='bold'>Hello world</font>"`
start = 0, end = 11 is 
`Hello world` or `<font famil`?

Reply:
as the font is a style not text when markup enabled , it will not work on it.
so if start = 0, end ==11 it will be `Hello world`